tests/test_timestamp.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at:
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+
+# Python 2/3 compatibility
+from decimal import Decimal
+
+import pytest
+
+from amazon.ion.core import Timestamp, TimestampPrecision, record
+from tests import parametrize, listify
+
+
+class _P(record('timestamps', 'expected_value')):
+    def __str__(self):
+        return self.desc
+
+
+MISSING_MICROSECOND = [
+    (Timestamp(
+        2011, 1, 1,
+        0, 0, 0, None,
+        precision=TimestampPrecision.SECOND, fractional_seconds=0
+        ), 0),
+    (Timestamp(
+        2011, 1, 1,
+        0, 0, 0, None,
+        precision=TimestampPrecision.SECOND, fractional_seconds=Decimal('0.123456')
+        ), 123456),
+    (Timestamp(
+        2011, 1, 1,
+        0, 0, 0, None,
+        precision=TimestampPrecision.SECOND, fractional_seconds=Decimal('0.123456789')
+        ), 123456),
+    (Timestamp(
+        2011, 1, 1,
+        0, 0, 0, None,
+        precision=TimestampPrecision.SECOND, fractional_seconds=Decimal('0.000123')
+        ), 123),
+    (Timestamp(
+        2011, 1, 1,
+        0, 0, 0, None,
+        precision=TimestampPrecision.SECOND, fractional_seconds=Decimal('0.123000')
+        ), 123000)
+]
+
+
+MISSING_FRACTIONAL_PRECISION = [
```

Comment:
This should be `MISSING_FRACTIONAL_SECONDS`, right?

Reply:
Right. Changed.